### PR TITLE
Jonmv/deployment risk 2

### PR DESCRIPTION
**Pull request by jonmv**
@freva please review. This builds on #22076

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/RunSerializer.java [hunk 1]
```diff
@@ -174,7 +174,6 @@ private ApplicationVersion applicationVersionFrom(Inspector versionObject) {
             return ApplicationVersion.unknown;
 
         long buildNumber = versionObject.field(buildField).asLong();
-        // TODO jonmv: Remove source revision
```
Comment (freva) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/RunSerializer.java:177]: Why not?

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelperTest.java [hunk 2]
```diff
@@ -189,17 +191,16 @@ public void testResponsesWithDryRunDeployment() {
...
-
     private void assertResponse(HttpResponse response, String fileName) {
         try {
             Path path = Paths.get("src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/responses/").resolve(fileName);
-            String expected = Files.readString(path);
-            compare(response, expected);
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            response.render(baos);
+            byte[] actualJson = SlimeUtils.toJsonBytes(SlimeUtils.jsonToSlimeOrThrow(baos.toByteArray()).get(), false);
+            Files.write(path, actualJson);
```
Comment (freva) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelperTest.java:200]:
```suggestion
```

File: vespajlib/src/main/java/com/yahoo/slime/JsonFormat.java [hunk 3]
```diff
@@ -17,26 +22,31 @@
...
-    private final boolean compact;
+    private final int indent;
+
+    public JsonFormat(int indent) {
+        this.indent = requireInRange(indent, "JSON space indent count", 0, 8);
+    }
+
     public JsonFormat(boolean compact) {
-        this.compact = compact;
+        this(compact ? 0 : 1);
```
Comment (freva) [inline vespajlib/src/main/java/com/yahoo/slime/JsonFormat.java:32]: Change this to be the same as in `SlimeUtils` (2)?

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelperTest.java (hunk 2 above)
Comment (jonmv) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelperTest.java:200]:
```suggestion
            // Files.write(path, actualJson);
```

File: vespajlib/src/main/java/com/yahoo/slime/JsonFormat.java (hunk 3 above)
Comment (jonmv) [reply vespajlib/src/main/java/com/yahoo/slime/JsonFormat.java:32]: Maybe. 1 is just a bad choice. Didn't dare do it in fear o breaking test data, but then again I already changed empty arrays/objects to render on one line, so perhaps ...

**Review by freva — APPROVED** [review]

**Comment by jonmv** [comment]
This requires a fix to the revision id copying.

**Pull request merged**